Broken GUI build due to missing linking against pthread

Since the recent change towards an object library for the core library I get a linking error when building the GUI:

> [ 91%] Linking CXX executable signal-estimator-gui
/usr/bin/ld: CMakeFiles/base_objects.dir/src/base/processing/CorrelationLatencyEstimator.cpp.o: undefined reference to symbol 'sem_destroy@@GLIBC_2.2.5'
/usr/bin/ld: /lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/signal-estimator-gui.dir/build.make:232: signal-estimator-gui] Error 1
make[1]: *** [CMakeFiles/Makefile2:306: CMakeFiles/signal-estimator-gui.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

OS: Debian Bullseye
gcc: 10.2.1

It can be fixed when linking the GUI against `${CMAKE_THREAD_LIBS_INIT}` (as it is also done for building the CLI)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -207,6 +207,7 @@ if(BUILD_GUI)
   target_link_libraries(signal-estimator-gui
     generated_objects
     base_objects
+    ${CMAKE_THREAD_LIBS_INIT}
     ${QWT_LIBRARY}
     Qt5::Core
     Qt5::Widgets

